### PR TITLE
Use writeThrough to avoid unnecessary MR stages (again)

### PR DESCRIPTION
**Pull request by toddmowen**

### CODE REVIEW

File: scalding/src/main/scala/commbank/coppersmith/scalding/HiveSupport.scala [hunk 1]
```diff
@@ -75,10 +75,20 @@ object HiveSupport {
...
+          written     <- partitioned.writeThrough(sink).withSubConfig(createUniqueFilenames(_))
+                         // Performance note: writeThrough works by returning the sink as a new source.
+                         // Unfortunately, this means an extra MapReduce step involving a scan of all the
+                         // data just written in order to obtain a list of partitions, which cascading
+                         // actually knows already (see BasePartitionTap.getChildPartitionIdentifiers).
+          oPValues    <- written.aggregate(Aggregator.toSet.composePrepare(_._1)).toOptionExecution
+                         .recoverWith {
+                           // Another writeThrough quirk: we need to explicitly handle the edge case where
+                           // no features were written (hence no partitions).
+                           case ex: java.io.IOException if ex.getMessage == "No input paths specified in job"
```
Comment (dkristian) [inline scalding/src/main/scala/commbank/coppersmith/scalding/HiveSupport.scala]: How stable is this string? I don't suppose there's a more specific exception we can catch instead?

**Comment by dkristian** [comment]
👍

File: scalding/src/main/scala/commbank/coppersmith/scalding/HiveSupport.scala (hunk 1 above)
Comment (toddmowen) [inline scalding/src/main/scala/commbank/coppersmith/scalding/HiveSupport.scala]: At least the tests cover this case, albeit not explicitly. (It was only because of a failing test that I discovered this edge case).
